### PR TITLE
Use `u64_digit` for 128-bit targets

### PR DESCRIPTION
**Pull request by icedrocket**
`CARGO_CFG_TARGET_*` variables are always set when cargo calls the build script. If these variables are not set, it should be considered a bug.

This PR makes sure these environment variables aren't missing, and cleans up the code a bit.

### CODE REVIEW

**Comment by cuviper** [comment]
> If these variables are not set, it should be considered a bug.

I'm not sure why that's so. It seems fine to me to make a conservative choice in case they're missing, for whatever reason.

**Comment by icedrocket** [comment]
Yes, there is no rule that it must be specified, but most crates call unwrap when loading Cargo-related environment variables. Since we're only supporting crates through the Cargo package manager, we're better off taking a conservative approach.

**Comment by cuviper** [comment]
Note that we both said "conservative" with different intent...

What is the *benefit* to being stricter about this? We risk breaking the build, however unlikely that may be, but there's no point in doing that when the code will still work fine with a fallback.

**Comment by icedrocket** [comment]
I checked when these environment variables were added to Cargo. These variables were added in [late 2016](https://github.com/rust-lang/cargo/commit/23587ee744ebd0f85714529f9c956b10803c4be8), after Rust 1.0 was released. This is before Rust 1.14 was released, so these environment variables are guaranteed to be set in Rust 1.31 and later.

As you mentioned, not being strict about missing environment variables doesn't currently cause any problems building crates. However, the reason I created this PR was because missing environment variables could be a bug in Cargo. AFAIK, there is no other program other than Cargo that uses a build script written in Rust.

File: build.rs [hunk 1]
```diff
@@ -5,29 +5,31 @@ use std::io::Write;
 use std::path::Path;
 
 fn main() {
-    let pointer_width = env::var("CARGO_CFG_TARGET_POINTER_WIDTH");
-    let u64_digit = pointer_width.as_ref().map(String::as_str) == Ok("64");
+    let ptr_width = &*env::var("CARGO_CFG_TARGET_POINTER_WIDTH").unwrap();
+
+    let u64_digit = ["64", "128"].contains(&ptr_width);
```
Comment (cuviper) [inline build.rs:10]: FWIW, I do agree that 128-bit targets should use `u64_digit` -- that's a positive change we can make. (even though there are no such targets yet, AFAIK)

Comment (icedrocket) [reply build.rs:10]: I'll create a new PR to reflect what you mentioned.

**Comment by icedrocket** [comment]
Now, thinking back to what I did a few weeks ago, I tried to keep my code as concise as possible. A syntax included in the latest Rust was not included in Rust 1.31, making it difficult to express in one line. For example:

```rust
let ptr_width = env::var("CARGO_CFG_TARGET_POINTER_WIDTH").ok();

let u64_digit = matches!(ptr_width.as_deref(), Some("64" | "128"));

// ...
```

```rust
let arch = env::var("CARGO_CFG_TARGET_ARCH").ok();

if let Ok(arch @ ("x86_64" | "x86")) = arch.as_deref() {
    //...
}
```

I couldn't use these patterns to continue supporting version 1.31, so I thought about how to express them concisely. One of the methods I've used to do this is just unwrap the Result. If we don't unwrap the Result, the expression becomes longer:

```rust
let u64_digit = ptr_width
    .as_ref()
    .map_or(false, |x| ["64", "128"].contains(&&**x));
```

or

```rust
let u64_digit = match ptr_width.as_ref().map(|x| &**x) {
    Some("64" | "128") => true,
    _ => false,
};
```

**Comment by cuviper** [comment]
There's no need to golf this, especially since it shouldn't need much change. Either of those longer options look fine, although you'll have to split that pattern too, `Some("64") | Some("128")`.

**Comment by icedrocket** [comment]
I am trying to add a new commit, but it is not reflected in the pull request. I contacted GitHub Support and am waiting for a response.

**Comment by icedrocket** [comment]
I'll just create a new pull request.

**Comment by icedrocket** [comment]
I'll just create a new pull request.

**Pull request closed without merging**